client/components/AdminDashboard.jsx

```diff
@@ -0,0 +1,108 @@
+"use client";
+import React, { useState } from "react";
+import { DataGrid } from "@mui/x-data-grid";
+
+export default function AdminDashboard() {
+  //   const [hasVisited, setHasVisited] = useState(false);
+  const status_options = ["Requested", "Confirmed", "Pending", "Visited"];
+
+  const [rows, setRows] = useState([
+    {
+      id: 1,
+      col1: false,
+      col2: "Confirmed",
+      col3: "6a-8a",
+      col4: "Targaryen",
+      col5: "Daenerys",
+      col6: "333-333-3333",
+      col7: "iConquor@westeros.com",
+      col8: "333 Dothraki Valley, Free City of Penthos, Valerya 33333",
+    },
+    {
+      id: 2,
+      col1: false,
+      col2: "Pending",
+      col3: "2p-4p",
+      col4: "Lannister",
+      col5: "Tyrion",
+      col6: "031-294-4822",
+      col7: "iDrink@andKnowThings.com",
+      col8: "699 Hand of the King Row, Here and There, Seven Kingdoms 00000",
+    },
+    {
+      id: 3,
+      col1: false,
+      col2: "Requested",
+      col3: "11a-1p",
+      col4: "Stark",
+      col5: "Arya",
+      col6: "909-485-3822",
+      col7: "aGirl@IsNoOne.com",
+      col8: "39842 Northern Way, Many Faced God, Esos 10101",
+    },
+  ]);
+
+  const columns = [
+    {
+      field: "col1",
+      headerName: "Mark as visited",
+      //   type: "boolean",
+      width: 150,
+      renderCell: (params) => (
+        <input
```

Comment:
This feels like added functionality to allow admins to unvisit and appointment. By my reading of our stories this wasn't included? If we keep it as a one-way action, then a `<Button/>` is better suited.

I might argue that a button is better anyway, as with a checkbox, there's little feedback to the user than an async action is being performed. Usually sites might include a toast or something to reassure a user that they're change is being saved.

Reply:
<img width="998" alt="Screenshot 2024-11-14 at 9 57 25 PM" src="https://github.com/user-attachments/assets/928b40fe-d0ea-48b1-9e95-e4748730ab97">
The last item under Admin Interface is what I was referencing when I wrote this in.

It's not actually to unvisit an appointment, but for admin to mark an appointment as visited.
I put in the uncheckmark condition in case the checkmark was accidentally clicked and it had to be reverted back. I started with a checkbox because there was a  checkmarked option built into x-grid. But then I ended up needing to customize it anyway. I'm not tied to the idea of keeping the checkbox so don't mind changing it to a button if you think that's better.